### PR TITLE
Enhance the measure query

### PR DESCRIPTION
**Pull request by hanahmily**
Introduced a series of enhancements:

* Add paging operation: limit, offset and order by
* Default time range criterion of the stream querying to [0, maxnanosecond]

Signed-off-by: Gao Hongtao <hanahmily@gmail.com>

### CODE REVIEW

File: api/proto/banyandb/measure/v1/query.proto [hunk 1]
```diff
@@ -90,4 +90,10 @@ message QueryRequest {
   } 
   // top limit the result based on a particular field
   Top top = 9;
+  // offset is used to support pagination, together with the following limit
+  uint32 offset = 10;
```
Comment (lujiajing1126) [inline api/proto/banyandb/measure/v1/query.proto:94]: But what's the difference between `top` and `limit/orderBy`?

File: pkg/timestamp/nano.go [hunk 2]
```diff
@@ -69,6 +72,7 @@ var (
 
 	ErrTimeOutOfRange     = errors.Errorf("time is out of range %d - %d", MinNanoTime, MaxNanoTime)
 	ErrTimeNotMillisecond = errors.Errorf("time is not millisecond precision")
+	ErrTimeEmpty          = errors.Errorf("time is empty")
```
Comment (lujiajing1126) [inline pkg/timestamp/nano.go:75]: These errors should contain gRPC status codes. Currently client cannot recognize these error types.

**Review by lujiajing1126 — CHANGES_REQUESTED** [review]

File: api/proto/banyandb/measure/v1/query.proto (hunk 1 above)
Comment (wu-sheng) [reply api/proto/banyandb/measure/v1/query.proto:94]: It seems we should merge these two?

Comment (hanahmily) [reply api/proto/banyandb/measure/v1/query.proto:94]: `top` operation is based on `field` while `limit/order by` is on `tag`.  They are different.

Comment (lujiajing1126) [reply api/proto/banyandb/measure/v1/query.proto:94]: > `top` operation is based on `field` while `limit/order by` is on `tag`. They are different.

Oh, I see. Then which one has high priority? Or they are exclusive?

Comment (hanahmily) [reply api/proto/banyandb/measure/v1/query.proto:94]: We can apply them at the same time. `limit` returns the results based on `top`'s output.

Comment (hanahmily) [reply api/proto/banyandb/measure/v1/query.proto:94]: I will add more details in the comments.

File: pkg/timestamp/nano.go (hunk 2 above)
Comment (hanahmily) [reply pkg/timestamp/nano.go:75]: done

File: api/proto/banyandb/measure/v1/query.proto (hunk 1 above)
Comment (hanahmily) [reply api/proto/banyandb/measure/v1/query.proto:96]: done

Comment (lujiajing1126) [reply api/proto/banyandb/measure/v1/query.proto:96]: Normally, we don't use topN mixed with limit/offset right?

**Comment by lujiajing1126** [comment]
After this PR is merged, shall we pass a default time range in the client? The logic is moved to the server, right?

File: api/proto/banyandb/measure/v1/query.proto (hunk 1 above)
Comment (hanahmily) [reply api/proto/banyandb/measure/v1/query.proto:96]: Exactly. We don't have such a use case in OAP integration.

**Comment by hanahmily** [comment]
> After this PR is merged, shall we pass a default time range in the client? The logic is moved to the server, right?

Yes, the server will provide a default range.

**Review by lujiajing1126 — APPROVED** [review]
LGTM now

**Pull request merged**